db/db_impl.cc

```diff
@@ -3417,6 +3419,13 @@ Status DBImpl::GetLatestSequenceForKey(SuperVersion* sv, const Slice& key,
     return Status::OK();
   }
 
+  SequenceNumber min_seq_in_mem = sv->mem->GetEarliestSequenceNumber();
+  if (min_seq_in_mem != kMaxSequenceNumber &&
+      min_seq_in_mem <= lower_bound_seq) {
```

Comment:
I dont think this optimization is correct. Consider min_seq_in_mem <= lower_bound_seq < seq. In that case we have to return the seq as it is a conflict.

Reply:
Nevermind. I just noticed this is after we already checked the memtable content.

Reply:
I think we should reduce <= to <.
If lower_bound_seq is snapshot seq then <= makes sense. But if it is min_uncommitted, then returning not found here if min_seq_in_mem == min_uncommitted would cause a bug. This is because a value with a seq equal to min_uncommitted could later be committed with a seq larger than the snapshot seq so it needs to be returned for conflict checking.

Reply:
Is this comment applied?